rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,535 @@
+
+        An example of such transaction is:
+        https://etherscan.io/tx/0x5a7849ab4b7f7de2b005deddef24a094387c248c3bcb06066109bd7852c1d8af
+        """
+        # check if ETH is sent to sai proxy contract or a proxy and mark it
+        # as a deposit since the proxy handles the conversion to WETH
+        # and transfer to the sai tub contract
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.counterparty == MAKERDAO_SAI_PROXY_CONTRACT and
+                event.asset == self.eth
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Supply {event.balance.amount} {self.eth.symbol} to SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                return None, []
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == self.eth
+            ):
+                # check for PETH mint event to match the deposit
+                for log in all_logs:
+                    if (
+                        log.topics[0] == PETH_MINT_EVENT_TOPIC and
+                        log.address == POOLED_ETHER_ADDRESS and
+                        hex_or_bytes_to_int(tx_log.topics[2]) == hex_or_bytes_to_int(log.data[:32])
+                    ):
+                        event.event_type = HistoryEventType.DEPOSIT
+                        event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                        event.notes = f'Supply {event.balance.amount} {self.eth.symbol} to SAI vault'  # noqa: E501
+                        event.counterparty = CPT_SAI
+                        return None, []
+
+        return None, []
+
+    def _maybe_enrich_sai_tub_transfers(
+            self,
+            token: EvmToken,  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+    ) -> bool:
+        """This method enriches relevant asset transfers to & from the SaiTub contract."""
+        if (
+            event.event_type == HistoryEventType.SPEND and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.counterparty in (MAKERDAO_SAITUB_CONTRACT, MAKERDAO_SAI_PROXY_CONTRACT)
+        ):
+            if event.asset == self.weth:
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Supply {event.balance.amount} WETH to SAI vault'
```

Comment:
here and in other places you seem to use direct symbols, while in others you use the `.symbol` property. Please be consistent. Use `.symbol` everywhere. I won't tell you every single place. Check this file please.